### ISSUE TITLE
Linux support please.

### ISSUE THREAD

**Issue by AndKe** [praydog/UEVR#280]
subject says everything.

**Comment by DocMAX** [comment]
+1 - The world is going linux

**Comment by Patola** [comment]
I've read that it works on Linux, but the procedure is certainly roughly documented. It would be great to have a space in UEVR's own tutorials.

**Comment by stephensrmmartin** [comment]
> I've read that it works on Linux, but the procedure is certainly roughly documented. It would be great to have a space in UEVR's own tutorials.

Yes it does work, though support is not 1:1 with Windows. For example, of all UE games I have tried (6 or so), all but one have worked seamlessly with UEVR. Satisfactory is currently crashing for me upon injection.

For future reference, these steps work on linux (using proton via steam):

1. Use protontricks -s to find the appid of your game.
2. Install dotnet 6 via protontricks: `protontricks-launch --no-bwrap --appid <appid here> --no-background-wineserver /path/to/dotnet_installer.exe
3. Launch uevr: `protontricks-launch --no-bwrap --appid <appid here> --no-background-wineserver /path/to/uevr.exe`
4. Import profile
5. Close uevr
6. Launch the game via Steam
7. Once loaded in, alt-tab out, and launch uevr again (see command above)
8. Wait until it detects the exe, and inject.

Make sure you choose openxr if using wivrn/envision, or openvr if using alvr.

**Comment by Patola** [comment]
Important note for people using UEVR: I first tried running uevr (which is now called `UEVRInjector.exe`) in a directory that was not in the prefix, e.g. `/dados/uevr` -- it didn't work. Moving this directory to the prefix, e.g. `~/.local/share/Steam/steamapps/compatdata/399810/pfx/drive_c/uevr`, and calling it from there, made it work:
`protontricks-launch --no-bwrap --appid 399810 --no-background-wineserver /home/patola/.local/share/Steam/steamapps/compatdata/399810/pfx/drive_c/uevr/UEVRInjector.exe`

Gameplay here, btw https://youtu.be/7DWyXGcPyWg

**Comment by stephensrmmartin** [comment]
@Patola Could you elaborate on "didn't work"?

Did it not open at all? Did it open but not detect an exe? Did it detect an exe but fail to inject? Did it inject but crash? Etc

I was able to inject in many games without issue regardless of the exe location, but I'm wondering if there's some funniness for some games (like satisfactory in my case, it fails to inject and just freezes video)

**Comment by Patola** [comment]
It said it couldn't find the file. I reckon it couldn't map `/dados/uevr/UEVRInjector.exe` to `z:\\dados\\uevr\UEVRInjector.exe` or similar path that the utility saw internally.

**Comment by Patola** [comment]
I've been noticing a pattern here that when I download an UEVR Profile from https://uevr-profiles.com and apply it to the UEVR in my games, it doesn't seem to work, at least not completely, The input do seem to work, but the 6DOF by making the arm disappear and the gun follow the hand motions do not seem to apply. Maybe that's due to the `uobjecthook` subdirectory in the zip file being all lowercase? Or any other reason?

Attaching the `log.txt` file from `AppData/Roaming/UnrealVRMod/Indiana-Win64-Shipping/log.txt` that seems to pertain to UEVR initializing from my last attempt with The Outer Worlds: Spacer's Choice Edition (appid 1920490). But I have this same issue in practically every game, like Trepang², Borderlands 3, Blair Witch and others.

[log.txt](https://github.com/user-attachments/files/21029751/log.txt)

**Comment by minyor** [comment]
Hello, I am too trying to use UEVR.
But every time I try to inject, it says:
**VR_init failed installation path could not be located (110)**

Steam games works fine
Should I somehow point UEVR to steamvr binary path?

**Comment by Patola** [comment]
> Hello, I am too trying to use UEVR on linux. I use ALVR and Quest 3 But every time I try to inject, it says: **VR_init failed installation path could not be located (110)**
> 
> Steam games works fine Should I somehow point UEVR to steamvr binary path?

You have to provide all useful data for people to help, like the game you're trying, what did you do previously, the command you're running, your distro, knowing your GPU can be helpful, even details like the filesystem type (btrfs, ext4, ntfs etc.) can sometimes help. Do not spare any details. Collecting and pasting the UEVR log like I did can help too.

**Comment by Patola** [comment]
> Here is how I start UEVR (using game's wineprefix):
> 
> > WINEFSYNC=1 WINEPREFIX=/data/Downloads/wine /home/minyor/.local/share/lutris/runners/proton/proton-cachyos-9.0-20250426-slr-x86_64_v3/files/bin/wine64 /data/Downloads/VR/UEVR/UEVRInjector.exe

This way of starting UEVR is incorrect. 
- WINEPREFIX is incorrect, since you are using proton
- It's better to start it via protontricks, not manually; and
- as I've mentioned in my messages here, there are directories outside of the game tree that it can't access. For this exact reason I have been copying the uevr directory to the game's wineprefix and starting if from there (via protontricks), like `~/.local/share/Steam/steamapps/compatdata/1778840/pfx/drive_c/uevr`, and then start UEVR from there after having started the game:
`protontricks-launch --no-bwrap --appid 1778840 --no-background-wineserver ~/.local/share/Steam/steamapps/compatdata/1778840/pfx/drive_c/uevr/UEVRInjector.exe`

**Comment by minyor** [comment]
This is a non-steam game, it doesn't have --appid
Tried to put UEVR folder inside of drive_c folder in game prefix and run from there but no change
If I try to start UEVR from lutris, like with "Run EXE inside Wine prefix" option or with "Open Wine console" and start UEVR from there, then UEVR for some reason fails to see the game pid and so Inject btb is inactive

**Comment by Patola** [comment]
> > Make sure you choose openxr if using wivrn/envision, or openvr if using alvr.
> 
> What if you're not using Envision? I only have the WiVRn flatpak installed.

It can work without envision (just the WiVRn flatpak), but it's like 100 times more awkward to do anything and you'll always be outdated. On envision you can add stuff like wlxoverlay-s to be able to see your screen and operate it in VR -- something that I find essential for these tasks (been using UEVR a lot lately). Why make your life more difficult? Use envision. It's open source and doesn't cost a dime.

**Comment by RaymondYouTube** [comment]
> > > Make sure you choose openxr if using wivrn/envision, or openvr if using alvr.
> > 
> > 
> > What if you're not using Envision? I only have the WiVRn flatpak installed.
> 
> It can work without envision (just the WiVRn flatpak), but it's like 100 times more awkward to do anything and you'll always be outdated. On envision you can add stuff like wlxoverlay-s to be able to see your screen and operate it in VR -- something that I find essential for these tasks (been using UEVR a lot lately). Why make your life more difficult? Use envision. It's open source and doesn't cost a dime.

I'm only asking because some Linux distros, such as SteamOS, Fedora Silverblue, and NixOS have an immutable file system. Now you can disable the read-only file system, however, whenever there's an update, like a SteamOS update for instance, anything you install outside of flatpak (via pacman for instance) may be wiped with the next SteamOS update.

**Comment by Patola** [comment]
Ok, at this point I got a playlist for my UEVR gameplays, 14 at this time and it will eventually grow to many dozen titles. Most of them have custom instructions for each game (say, when to run the injection, some games like Borderlands 3 require it to be very early to inject), although they follow the general recipe described here. The instructions are in the video descriptions and some of them have extra steps on a pinned comment. The specific versions of software I used (UEVR, WiVRn, etc.) are also shown. I hope it proves useful. https://www.youtube.com/watch?v=X_5FBRsRP-g&list=PLoRF2wMS15vV65fXuudiHHpYZlNMZpkQB

**Comment by RaymondYouTube** [comment]
> I use steamtinkerlaunch to make the process of using UEVR easier. Install stl as steam compat tool then select it as compat tool for the UEVR games. In the stl ui do the following:
> 
> 1. One time command: install the dotnet6 runtime
> 2. configure the game:
> 
> * Use custom command: UEVRInjector.exe
> * disable steam runtime for the custom command (otherwise it doesnt see the game process)
> * Inject custom command
> 
> 3. Save and play
>    These steps only have to be done once but for each game.

When you use Steam Tinker Launch, when disabling the runtime for the custom command, are you referring to the Steam Linux Runtime? Or are you referring to something else? I'm trying to test this with ACE COMBAT 7: SKIES UNKNOWN.

**Comment by wallcarpet40** [comment]
Hello,

Has anyone been able to get Jedi Survivor to work with Linux and UEVR? 

For me, the game crashes, when injecting UEVR. I'm just able to see the UEVR mod GUI pop up on my monitor. Most other games do work, even DX12 games, but this one is giving a hard time.

**Comment by Patola** [comment]
> Hello,
> 
> Has anyone been able to get Jedi Survivor to work with Linux and UEVR?
> 
> For me, the game crashes, when injecting UEVR. I'm just able to see the UEVR mod GUI pop up on my monitor. Most other games do work, even DX12 games, but this one is giving a hard time.

It helps if you say which profile you are using.

**Comment by wallcarpet40** [comment]
> It helps if you say which profile you are using.

Good call! I was not using any profile, just vanilla UEVR-nightly. I downloaded a profile from uevr-profiles.com and it works!
I wonder why it required that in the first place. Haven't needed that with any other game.

**Comment by victoragcosta** [comment]
> I use steamtinkerlaunch to make the process of using UEVR easier. Install stl as steam compat tool then select it as compat tool for the UEVR games. In the stl ui do the following:
> 
> 1. One time command: install the dotnet6 runtime
> 2. configure the game:
> 
> * Use custom command: UEVRInjector.exe
> * disable steam runtime for the custom command (otherwise it doesnt see the game process)
> * Inject custom command
> 
> 3. Save and play
>    These steps only have to be done once but for each game.

So, I did this with steamtinkerlaunch and it finally worked after days of trying with protontricks-launch and UEVR not seeing the game process. After that I saw that you disabled steam runtime and thought that was what I was doing wrong. I kinda don't want to have steamtinkerlaunch installed, so I tried adding `--no-runtime` to the protontricks-launch command and even tried with and without `--cwd-app` as a hail mary:

```
protontricks-launch --no-runtime --no-bwrap --no-background-wineserver --cwd-app --appid "$app_id" "$HOME/.steam/steam/steamapps/compatdata/$app_id/pfx/drive_c/users/steamuser/UEVR/$release/UEVRInjector.exe"
```

And it didn't work. I fail to see what's the difference between using steamtinkerlaunch and protontricks-launch.

Extra information:
- Game: Motortown: Behind The Wheel
- OS: NixOS
- Proton: GE-Proton10-34

If anyone has any information I would appreciate

**Comment by rhetz2002** [comment]
hey i hate to necro post but when i try to launch UEVR it freeses the moment it detects the game, iv tried moving it to the prefex and using both 
protontricks-launch --no-bwrap --appid <appid here> --no-background-wineserver /path/to/uevr.exe 

and another one 

WINEPREFIX=/home/USERNAME/.local/share/Steam/steamapps/compatdata/1332010/pfx WINEFSYNC=1 /home/USERNAME/.local/share/Steam/steamapps/common/Proton\ -\ Experimental/files/bin/wine /home/USERNAME/Downloads/UEVR/UEVRInjector.exe 

whenever i hover my mouse over the window it disappear and in order to close it i need to go into system monitor

been using proton experimental if that helps
